### ISSUE TITLE
Master React, Quick Code tutorial. Part 8

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/react-only-intern-23#9]
https://medium.com/quick-code/lets-learn-react-chapter-8-react-forms-f3137a5c07c7
## Expected result

- [ ] Read this tutorial chapter
- [ ] follow code steps
- [ ] move your code to this repository(at least one commit per day)
- [ ] explain to me, what did you learn from it in a few sentences(not less than 5)
- [ ] pass the test section
- [ ] after reviewing your progress and closing this issue - you can move to the next chapter.

## Test
- [ ] right now form looks very simple. 
And usually Meal form have more fields. So let's add them.

By using this database schema: https://github.com/ChickenKyiv/recipe-api-only/blob/master/Recipe-ChickenKyiv-Release%231%20Schema%20%20%20SqlDBM.png
We'll advance this form. I'm pretty sure that when you'll add all fields in this form, you'll realize that it's better to create a sub-components for each form field.
So, Meal is a similar Model to Recipe.
so we'll add all important things to that form.
* Each recipe has attributes: diets, allergies, etc. - text field for each one
* Each recipe has a weekday field(like today i have chicken, tomorrow i'll eat soup) - i assume it'll be checkbox.
* Nutritions is a sub-array with a few fields() - text fields
* Ingredients is a sub-array with a few ingredients inside(like chichen brests, salt, pepper, etc.) Let's have 4 ingredients per meal. 

And, make it works :) this is when a real ES6 is coming into the scene.
and don't forget that each field will also have an id field(later it can be useful, when you'll grab data from database)

#1

**Comment by atherdon** [comment]
For me personally, this is a most important chapter of this course, because it really starting to intersect with our main project...

**Comment by atherdon** [comment]
Btw, i think knowledge, that you gain in graphQL part about database schemes that we have will help you a bit here.

**Comment by sanchit94** [comment]
@atherdon Link to database schema has expired.

**Comment by atherdon** [comment]
Yeah, I've made big changes at the documentation, since i post this task. 
Please take a look here: https://chickenkyiv.gitbook.io/documentation/database-schemes/chickenkyiv-schema

**Comment by atherdon** [comment]
replied, thank you for helping me. a bit overwhelmed right now and can miss something

**Comment by sanchit94** [comment]
Hehe... I can understand as you have a lot of things to manage yourself.

**Comment by atherdon** [comment]
![xxx](https://media.giphy.com/media/l41m3Wa2np53KAr60/giphy.gif)

**Comment by sanchit94** [comment]
Something wrong with the link you gave.  It's not showing the image. Yesterday it was showing. Please check, or is this glitch just in my system?
![image](https://user-images.githubusercontent.com/33549324/42738896-b7871904-8840-11e8-9335-af17f477ea8b.png)

**Comment by atherdon** [comment]
fixed

**Comment by sanchit94** [comment]
Okay what do you suggest we stick to modals, or create a new page for adding meals? I feel just one modal is too much data at one place so it can be something like a wizard. Please suggest @atherdon

**Comment by atherdon** [comment]
i'm ok with modals, but separated page is more what i'll adore. main idea is to re-use your code later, so you like building a set of components that we'll export and then import as separated module into other projects

**Comment by sanchit94** [comment]
Alright! I too like the idea of creating a separate page for now and having modularised code because later we can do whatever we want with it! :)

**Comment by atherdon** [comment]
yeah - that's a plan. Btw, feel free if you have questions - maybe i can help you with this chapter

**Comment by sanchit94** [comment]
Well, I had a gazillion questions when I first looked at this, so my plan is -> Write code and push it -> Then we can make specific changes, and any of my question can be better addressed!

**Comment by atherdon** [comment]
ok, will wait for your commits


On 17-Jul-2018 at 09:17 PM, Sanchit Sharma wrote:


Well, I had a gazillion questions when I first looked at this, so my plan
is -> Write code and push it -> Then we can make specific changes, and any
of my question can be better addressed!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/9#issuecomment-405731155>,
or mute the thread

**Comment by sanchit94** [comment]
Just this question is coming to my mind, in` Ingredients.js`, I declare `multiVal` in the starting to capture data for the state variable `multivalue `and transfer it to parent `MealPlan.js` using `multiVal` variable. Is there a better way to do this, I used this approach just to make it work and it is 0% elegant. I cannot do `event.target.value` in `handleIngredients()` because that won't work.

**Comment by atherdon** [comment]
so if i understand you correctly - you want to pass value from child to
parent component?


On 17-Jul-2018 at 09:24 PM, Sanchit Sharma wrote:


Just this question is coming to my mind, inIngredients.js, I declare
multiVal in the starting to capture data for the state variable multivalueand
transfer it to parent MealPlan.js using multiVal variable. Is there a
better way to do this, I used this approach just to make it work and it is
0% elegant. I cannot do event.target.value in handleIngredients() because
that won't work.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/9#issuecomment-405733060>,
or mute the thread

**Comment by atherdon** [comment]
yeah, i find that great plugin while working on recipe search form. and it
really gamechanger. if you want - i can share it with you, so you'll be
able to take a quick look?


On 17-Jul-2018 at 09:41 PM, Sanchit Sharma wrote:


Great, so I won't worry about this for now and focus on this issue!! Thanks
:)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/9#issuecomment-405737420>,
or mute the thread

**Comment by sanchit94** [comment]
Sure, please provide me a link to that @atherdon

**Comment by atherdon** [comment]
check it github.com/final-form/final-form#-final-form


On 18-Jul-2018 at 09:07 AM, Sanchit Sharma wrote:


Sure, please provide me a link to that @atherdon
<https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/9#issuecomment-405863961>,
or mute the thread

**Comment by sanchit94** [comment]
Looks pretty cool. So when do you suggest we include this in our project?

**Comment by atherdon** [comment]
if you want - you can create a separate component and create a meal form
by using it. i didn't use it yet, but if oyu'll learn how to do it - then
you'll be able to help us with search form too!


On 18-Jul-2018 at 09:37 AM, Sanchit Sharma wrote:


Looks pretty cool. So when do you suggest we include this in our project?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/9#issuecomment-405872402>,
or mute the thread

**Comment by sanchit94** [comment]
Well, currently I'm working on adding all the additional fields according on the database schema on the meal and that is almost done. I'll push it and once that is finalised somehow then I'll use this to create form for Recipy!

**Comment by atherdon** [comment]
good! i like your progress. really want to start to use your code in our
main projects


On 18-Jul-2018 at 09:44 AM, Sanchit Sharma wrote:


Well, currently I'm working on adding all the additional fields according
on the database schema on the meal and that is almost done. I'll push it
and once that is finalised somehow then I'll use this to create form for
Recipy!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/9#issuecomment-405874516>,
or mute the thread
